### ISSUE TITLE
DL 0-xx button doesn't work

### ISSUE THREAD

**Issue by cwnevergg** [leoding86/webextension-pixiv-toolkit#39]
problems:
the **Generate GIF** and **Download GIF** is fully usable, 
but the button **DL 0-xx** doesn't work. 
extra information:
Before February 18,2020 , the DL 0 - xx is works well
Platform : OS X 10.15.3
Browser: Google Chrome 80.0.3987.116 64bit

**Comment by leoding86** [comment]
Some users find some works in Pixiv need access techorus-cdn.com recently but I can't find any kind of works. I added a setting in version 3.9.0 to manually grant the extension to access this domain. This version is still pending review by Chrome web store.

**Comment by cwnevergg** [comment]
> Some users find some works in Pixiv need access techorus-cdn.com recently but I can't find any that kind of works. So I added a setting in version 3.9.0 to manually grant the extension to access this domain. This version is still pending review by Chrome web store.

Thanks for your great work!For me,Generate GIF and Download GIF is fully usable(version 3.8.1).
Only DL 0-xx does not work.Waiting for version 3.9.0.

**Comment by hal0thane** [comment]
Thank you, leoding86, for all your work on this great extension. Unfortunately, the new version v3.9.0-2020022203 still doesn't seem to have cleared up the problem for me (I am "B H" on Google). After the extension updated, I went to settings and granted permission to the new domain. After that, it didn't work, so I tried hitting Reload, and still did not work.

This may be related: I just found this in the console. When I click the DL Image button, the following errors show up in the console:
```
Access to XMLHttpRequest at 'https://tc-pximg01.techorus-cdn.com/img-original/img/2020/02/23/01/37/06/79666245_p0.jpg' from origin 'https://www.pixiv.net' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
09:16:13.277 app.js:1 GET https://tc-pximg01.techorus-cdn.com/img-original/img/2020/02/23/01/37/06/79666245_p0.jpg net::ERR_FAILED
(anonymous) @ app.js:1
e @ app.js:1
value @ app.js:1
downloadButtonClicked @ app.js:1
click @ app.js:1
He @ vue.min.js:6
n @ vue.min.js:6
He @ vue.min.js:6
e.$emit @ vue.min.js:6
click @ app.js:1
He @ vue.min.js:6
n @ vue.min.js:6
Yr.o._wrapper @ vue.min.js:6
09:16:13.277 79666245:1 Uncaught (in promise) undefined
Promise.then (async)
downloadButtonClicked @ app.js:1
click @ app.js:1
He @ vue.min.js:6
n @ vue.min.js:6
He @ vue.min.js:6
e.$emit @ vue.min.js:6
click @ app.js:1
He @ vue.min.js:6
n @ vue.min.js:6
Yr.o._wrapper @ vue.min.js:6
```

I hope this is helpful information. If you would like the source code of an image page, I can share that with you as well.

**Comment by hal0thane** [comment]
Here's [the source I captured](https://gist.github.com/hal0thane/da7b85d45023c23bd43fde3b6c158394)
And here's how the relevant elements appear in the [element inspector](https://gist.github.com/hal0thane/c21a58c33fa0edaf82fa5b82e9e8f44e).
Hopefully some of that helps! If there's anything else I can help with, I'd be happy to.

**Comment by TestPolygon** [comment]
One more example: [mhtml page](https://github.com/TestPolygon/temp/raw/master/artworks_79716436.mhtml)

**Comment by leoding86** [comment]
Sorry, I declare the wrong domain in manifest. Fix in next version.

**Comment by hal0thane** [comment]
Thank you, leoding86. I was afraid it would be more complicated.

**Comment by ian-coog** [comment]
3.9.0 for me doesn't save to C:\temp\pixiv_downloads\ anymore, anything, not single images, not zipped collections or webm generated from ugoiras. 
The jpg/zip files gets correctly downloaded into the browser cache ("%localappdata%\Vivaldi\User Data\Default\Cache\" for me) but then not saved into my default download location (C:\temp\, and relative location in PixivToolkit settings is pixiv_downloads/)
3.8.2 worked yesterday.
I tried to change also the rename masks to {id}_{pageNum} or just {id} and nothing changes.

**Comment by ian-coog** [comment]
I've now reverted to https://github.com/leoding86/webextension-pixiv-toolkit/releases/tag/3.7 by packing the extension and dragging the crx in the extensions page, everything gets saved now.
Definately a regression bug.

**Comment by ian-coog** [comment]
I've also tried downloading the 3.8.2 crx from https://www.crx4chrome.com/crx/87183/
everything gets saved correctly. The bug is clearly in 3.90 at this point.

**Comment by leoding86** [comment]
Try update to 3.9.1 in Chrome.

**Comment by hal0thane** [comment]
Thank you, leoding86. I can confirm that the extension once again _works for me_. Yay! Thank you! I don't really use Patreon, but is there some other way that I can give you a one-time donation for your work? You always update it quickly when there is a need, along with new features from time to time, and I appreciate it. I use your extension almost every day, so I I would like to compensate you for it.

**Comment by ian-coog** [comment]
3.9.1 still doesn't work for me, I've even granted access to the cdn but nothing gets saved
These are my settings if it helps
``
{"MangaPageNumberStartWithOne":false,"accessTechorusCdn":false,"askDownloadSavedWork":true,"autoActivateDownloadPanel":true,"downloadRelativeLocation":"pixiv_downloads/","downloadSaveAs":false,"enableExtTakeOverDownloads":true,"enableExtend":false,"enableExtension":true,"enablePackUgoiraFramesInfo":true,"enablePtkSearch":true,"enableSaveVisitHistory":true,"enableWhenUnderSeconds":1,"extendDuration":3,"featureKnown":false,"guideShowed":true,"illustrationDownloadIfReady":true,"illustrationImageRenameFormat":"{author}-{title}-{id}_{pageNum}","illustrationKeepPageNumber":false,"illustrationPageNumberStartWithOne":true,"illustrationRenameFormat":"{author}-{title}-{id}","importantNoticeDisplayed":true,"mangaImageRenameFormat":"{author}-{title}-{id}_{pageNum}","mangaPackAndDownload":true,"mangaPageNumberStartWithOne":true,"mangaPagesInChunk":99,"mangaRenameFormat":"{author}-{title}-{id}","notSaveNSFWWorkInHistory":false,"novelIncludeDescription":false,"novelRenameFormat":"{title}-{id}","statIllustDownloaded":6,"statMangaDownloaded":0,"statNovelDownloaded":0,"statUgoiraDownloaded":2,"subscribedUsers":{},"ugoiraDisplayDownloadProgress":true,"ugoiraGenerateAndDownload":true,"ugoiraQuanlity":1,"ugoiraRenameFormat":"{author}-{title}-{id}","version":"3.9.1","visitHistoryType":"list"}
``

**Comment by hal0thane** [comment]
I can't say for certain, ian-coog, since I'm not the dev, but your second item there, `accessTechorusCdn` is set to `false`. Seems like it hasn't been set in fact.

**Comment by ian-coog** [comment]
hal0thane; yeah seems weird but remains false whatever if I grant it or not. In any case I seem to be able to download everything with 3.8.2 so I shouldn't be even needing that setting.

**Comment by MangoNation** [comment]
@leoding86 The new version works. Thank you again so much for adjusting this!

**Comment by leoding86** [comment]
@ian-coog Which browser are you using? If you are using Chrome, could you see the *://*.techorus-cdn.com/* in the "Site access" list from the extension detail after you grant the extension to access techorus-cdn.com?

**Comment by hal0thane** [comment]
One thing you might try, @ian-coog, is to export the settings like you did here, edit the file to change the `false` to `true` in the `accessTechorusCdn` key, and then reimport the settings. If that still doesn't do it, I would suggest reinstalling the extension. If you do that, you should probably reconfigure the settings manually instead of importing your settings.

**Comment by ian-coog** [comment]
@leoding86 : I'm using Vivaldi 2.11.1811.28
retried Pixiv Toolkit 3.9.1 (it auto updated from 3.8.2 after restarting the browser anyway)
I have granted the cdn access in the settings and now in the exported settings I have "accessTechorusCdn":true
Nothing changed, no files gets saved. But it's because I have "enableExtTakeOverDownloads":true and "downloadSaveAs":false which is "Extension takes over downloads" + "Ask where to save each file before downloading" disabled.
I have to disable "take over" and let the browser prompt for the download dialogue, then I can save the file wherever I want.
In other words, it's the autosave to the that got broken.

**Comment by ian-coog** [comment]
@leoding86 v3.10.0-2020030306 installed and now everything works perfectly, Extension takes over downloads, automatically saves in relative location single images, zipped collections and ugoira->webm. Thanks!
